website/profile/views.py

```diff
@@ -85,20 +86,64 @@ def update_user(auth):
     data = request.get_json()
 
     # TODO: Expand this to support other user attributes
-    if 'timezone' in data:
-        if data['timezone']:
-            user.timezone = data['timezone']
+
+    if 'emails' in data:
+        # removals
+        removed_emails = [
+            each
+            for each in user.emails + user.unconfirmed_emails
+            if each not in [x['address'] for x in data['emails']]
+        ]
+
+        for address in removed_emails:
+            if address in user.emails:
+                user.emails.remove(address)
+            user.remove_unconfirmed_emails(address)
+
+        # additions
+        added_emails = [
+            each['address']
+            for each in data['emails']
+            if each['address'] not in user.emails
+            and each['address'] not in user.unconfirmed_emails
+        ]
+
+        for address in added_emails:
+            try:
+                user.add_unconfirmed_email(address)
+            except ValueError:
+                continue
+
+            # TODO: This setting is now named incorrectly.
+            if settings.CONFIRM_REGISTRATIONS_BY_EMAIL:
+                send_confirm_email(user, email=address)
+
+        # set username
+        username = None
+        try:
+            username = [each for each in data['emails']
+                        if each.get('primary')][0]['address']
+        except IndexError:
+            pass
+
+        # make sure the new username has already been confirmed
+        if username and username in user.emails:
+            user.username = username
+
     if 'locale' in data:
         if data['locale']:
             locale = data['locale'].replace('-', '_')
             user.locale = locale
+    if 'timezone' in data:
+        if data['timezone']:
```

Comment:
Yes, but that's outside the scope of this PR and I don't want to add it right now.

I'll be working in this method extensively next week; I've added a TODO and will take care of it then.